### ISSUE TITLE
[Feature Request] Prompt for confirmation by default for delete/terminate/cancel

### ISSUE THREAD

**Issue by lorensr** [temporalio/cli#289]
### Is your feature request related to a problem? Please describe.

User accidentally deleting a workflow.

### Describe the solution you'd like

By default, prompt something like:

```
$ temporal workflow delete -w foo
Id: foo
Would you like to X this Workflow? Y/n
```

Pass `-y` to skip prompt

**Comment by feedmeapples** [comment]
we can add this also for consistency with batch operations such as `temporal workflow terminate --query ..`

Then users could disable prompt for default env with `temporal env set default.yes true`

**Comment by bergundy** [comment]
I don't think we want confirmation for termination / cancellation of a single workflow, it's reversible.
I'm okay with adding it to delete.

**Comment by lorensr** [comment]
How do you reverse, by resetting to the last event before termination/cancellation? Wonder how many know that. I guess those that ask support will find out.

**Comment by lorensr** [comment]
I guess since reset will be in the UI for closed workflows, it will be much more discoverable.
